src/nemos/regularizer.py

```diff
@@ -336,156 +230,62 @@ def _penalization(
+        def _penalized_loss(params, X, y):
             return loss(params, X, y) + self._penalization(params)
 
-        return super().instantiate_solver(penalized_loss, *args, **kwargs)
-
+        return _penalized_loss
 
-class ProxGradientRegularizer(Regularizer, abc.ABC):
-    """
-    Abstract class for ptimization solvers using the Proximal Gradient method.
-
-    This class utilizes the `jaxopt` library's Proximal Gradient optimizer. It extends
-    the base Solver class, with the added functionality of a proximal operator.
-
-    Attributes
-    ----------
-    allowed_solvers : List[...,str]
-        A list of solver names that are allowed to be used with this regularizer.
-    """
-
-    _allowed_solvers = ("ProximalGradient",)
-
-    def __init__(
-        self,
-        solver_name: str,
-        solver_kwargs: Optional[dict] = None,
-        regularizer_strength: float = 1.0,
-        **kwargs,
-    ):
-        super().__init__(solver_name, solver_kwargs=solver_kwargs)
-        self.regularizer_strength = regularizer_strength
-
-    @abc.abstractmethod
-    def _get_proximal_operator(
-        self,
+    def get_proximal_operator(
+            self,
     ) -> ProximalOperator:
-        """
-        Abstract method to retrieve the proximal operator for this solver.
-
-        Returns
-        -------
-        :
-            The proximal operator, which typically applies a form of regularization.
-        """
-        pass
-
-    def instantiate_solver(
-        self, loss: Callable, *args: Any, **kwargs: Any
-    ) -> Tuple[SolverInit, SolverUpdate, SolverRun]:
-        """
-        Instantiate the solver with the provided loss function.
-
-        Instantiate the solver with the provided loss function and return callable functions
-        that initialize the solver state, update the model parameters, and run the optimization.
-
-        This method creates a solver instance from jaxopt library, tailored to the specific loss
-        function and regularization approach defined by the Regularizer instance. It also handles
-        the proximal operator if required for the optimization method. The returned functions are
-         directly usable in optimization loops, simplifying the syntax by pre-setting
-        common arguments like regularization strength and other hyperparameters.
-
-        Parameters
-        ----------
-        loss :
-            The original loss function to be optimized.
+        def prox_op(params, l2reg, scaling=1.0):
+            Ws, bs = params
+            l2reg /= bs.shape[0]
+            # if Ws is a pytree, l2reg needs to be a pytree with the same
+            # structure
+            if isinstance(Ws, (dict, FeaturePytree)):
+                struct = jax.tree_util.tree_structure(Ws)
+                l2reg = jax.tree_util.tree_unflatten(
+                    struct, [l2reg] * struct.num_leaves
+                )
+            return jaxopt.prox.prox_lasso(Ws, l2reg, scaling=scaling), bs
```

Comment:
```suggestion
            return jaxopt.prox.prox_ridge(Ws, l2reg, scaling=scaling), bs
```

Reply:
Here there is a thing we need to take care of. The ridge regularized models we are implementing for all solvers but ProximalGradient  minimize the following penalized loss

```
loss(coeff, intercept) - 0.5 * reg_strength * norm(coeff)**2 / len(coeff).
```
There is an 0.5 multiplying the penalization. This is convention, and it's the same as sklearn;
Here we need to make sure that the proximal operator also solves the same problem. It may be that the prox operator is minimizing 

```
loss(coeff, intercept) - reg_strength * norm(coeff)**2 / len(coeff).
```

by default, and we need to either divide by two the regularization strength before passing it to the solver, or there is a "scaling" parameter in jaxopt that we may need to set to 0.5. 

The thing to do to verify that we are solving the same problem with all algorithm is to fit a GLM with Ridge with both ProximalGradient and the LBFGS, and check that the params are "allclose". I would add a test for that.